### DISCUSSION TITLE
Error "Index exceeds array bounds" by using Miller-class

### DISCUSSION THREAD

**Discussion by lehu47** [mtex-toolbox/mtex#1315] [Ask Anything]
Hello,

after loading ebsd-data, i am creating the ipfKey-variable with the "ipfColorKey"-function. Afterwards I am trying to plot the orientation map of the ebsd-data with the "ipfKey.orientation2color"-function. While executing this I get an "Error in Miller (line 86)" as seen below. The same problem occurs while using given code from the MTEX-website, where is the example coming from. 
I've already tried to unistall/install the mtex-package and i am using MTEX V. 5.7.0.

Thanks for the help in advance!!

```
 plotx2east, plotb2east
mtexdata sharp

ebsd = ebsd('calcite');

ipfKey = ipfColorKey(ebsd);

close all;
plot(ebsd,ipfKey.orientation2color(ebsd.orientations))
 
ebsd = EBSD (show methods, plot)
 
 Phase  Orientations     Mineral         Color  Symmetry  Crystal reference frame
     0    32 (0.16%)  notIndexed                                                 
     1  20119 (100%)     calcite  LightSkyBlue      -3m1       X||a*, Y||b, Z||c*
 
 Properties: x, y
 Scan unit : um
 
Index exceeds array bounds.

Error in Miller (line 86)
      m.CSprivate = getClass(varargin{4},'crystalSymmetry',[]);

Error in vector3d/rotate (line 57)
    v = Miller(v,q.SS);

Error in  .*  (line 32)
  r = rotate(b,a);

Error in ipfColorKey/orientation2color (line 63)
      h = inv(ori) .* oM.inversePoleFigureDirection;
```

**Comment by kilir** [comment]
Hi,
I cannot reproduce this error. What Matlab version are you using? What happens when you try the current mtex version?
Cheers,
Rüdiger

**Reply by lehu47** [reply]
Hi Rüdiger,

as far as I can see i am using the current mtex version, or is there a newer one than 5.7.0? My Matlab Version is 2018b, I will try to install the newest one and let you know!
Cheers,
Lennart

**Comment by ralfHielscher** [comment]
Hi,

for some reason there is a error in your source code, i.e., in the file mtex/geometry/@Miller/Miller.m

you have in line line 86
```matlab
m.CSprivate = getClass(varargin{4},'crystalSymmetry',[]);
```
but it should be 
```matlab
m.CSprivate = getClass(varargin,'crystalSymmetry',[]);
```
Maybe you simply reinstall MTEX?

Ralf.
